articles/openshift/oshift-netpol.md

```diff
@@ -0,0 +1,138 @@
+
+## Interacting with NetworkPolicy objects
+
+NetworkPolicy objects can be interacted with on the command line using the full name networkpolicy or short name netpol, for example:
+
+``` oc get netpol ``` - List NetworkPolicy objects in the current project.
+
+``` oc describe netpol <name> ``` - Provides more detailed information on the NetworkPolicy object.
+
+``` oc get netpol <name> -o yaml ``` - Provides yaml output of the NetworkPolicy object. Useful to reference existing rules when creating new objects.
+
+## Example of connecting services from two different projects
+
+NetworkPolicy objects are scoped at the project level. This means if you create a NetworkPolicy object using a template it will be created inside the project you're in and the rules will apply to pods within that project, with the exception of remote projects you may be referencing inside the rule. The object without any rules will look like the following:
+
+```
+apiVersion: extensions/v1beta1
+kind: NetworkPolicy
+metadata:
+  name: <desired_name_of_object>
+  namespace: <desired_project> (if not provided this will default to the project you are in when you run oc create)
+spec:
+```
+
+Inside the spec key we'll pass in the rules that will determine what our NetworkPolicy object does. The first thing you need to consider is the pods you want to target, this is done using the podSelector key. 
+You can use this to match all pods in a project or alternatively to match all pods with a certain label. If you leave this key blank it will match all pods. If you want to match pods with the label 'role=webserver' 
+it would look like:
+
+```
+apiVersion: extensions/v1beta1
+kind: NetworkPolicy
+metadata:
+  name: webserver-egress 
+  namespace: webserver
+spec:
+  podSelector:
+    matchLabels:
+      role: webserver
+```
+
+At this point we've matched the pods inside our webserver project that have the label role=webserver. Let's say for example we want to allow these pods to connect to a service inside a project we've named 'database' we could
+achieve this using a namespace selector, which evaluates projects based on a label. In this case we are going to be looking for the namespace label database=true. Firstly we need to ensure this label is applied using the 
+following command:
+
+> [!NOTE]
+> The cluster administrator role is required to add labels to namespaces.
+
+``` oc label namespace database database=true ```
+
+Our netpol object will now look like:
+
+```
+apiVersion: extensions/v1beta1
+kind: NetworkPolicy
+metadata:
+  name: webserver-egress
+  namespace: webserver
+spec:
+  podSelector:
+    matchLabels:
+      role: webserver
+  egress:
+  - to:
+    - namespaceSelector:
+        matchLabels:
+          database: "true"
+```
+
+We now have a template for a NetworkPolicy object that will match any pods labeled role=webserver inside the webserver project and allow them to communicate out to any projects labeled database=true. This is only half of the work,
```

Comment:
We now have a template for a NetworkPolicy object that matches any pods labeled `role=webserver` inside the webserver project and allows them to communicate out to any projects labeled `database=true`. This is only half the work; we still need the ingress allowance inside the database project. Our example template for that object looks like the following: